homeassistant/components/mqtt/config_flow.py

```diff
@@ -572,6 +595,44 @@ def validate_light_platform_config(user_data: dict[str, Any]) -> dict[str, str]:
     },
 }
 PLATFORM_MQTT_FIELDS = {
+    Platform.BINARY_SENSOR.value: {
+        CONF_STATE_TOPIC: PlatformField(
+            selector=TEXT_SELECTOR,
+            required=True,
+            validator=valid_subscribe_topic,
+            error="invalid_subscribe_topic",
+        ),
+        CONF_VALUE_TEMPLATE: PlatformField(
+            selector=TEMPLATE_SELECTOR,
+            required=False,
+            validator=cv.template,
+            error="invalid_template",
+        ),
+        CONF_PAYLOAD_OFF: PlatformField(
+            selector=TEXT_SELECTOR,
+            required=False,
+            validator=str,
+            default=DEFAULT_PAYLOAD_OFF,
+        ),
+        CONF_PAYLOAD_ON: PlatformField(
+            selector=TEXT_SELECTOR,
+            required=False,
+            validator=str,
+            default=DEFAULT_PAYLOAD_ON,
+        ),
+        CONF_EXPIRE_AFTER: PlatformField(
+            selector=EXPIRE_AFTER_SELECTOR,
+            required=False,
+            validator=cv.positive_int,
+            section="advanced_settings",
+        ),
+        CONF_OFF_DELAY: PlatformField(
+            selector=EXPIRE_AFTER_SELECTOR,
```

Comment:
The selector should be renamed to take into account that it is used on more places than just expire_after.

Reply:
Why is that needed?

Reply:
It is not necessary, just a nitpick. One may find it confusing to have re-use the selector for expire_after for another field.

Reply:
Renamed to TIMEOUT_SELECTOR